File: src/main/java/com/yassine/backend/Dao/Reclamation.java
package com.yassine.backend.Dao;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Reclamation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idReclamation;

    private Date date;
    private String description;
    private String type;
    private String status;

    @ManyToOne
    private Utilisateur utilisateur;

}

File: frontend/lib/models/Reclamation.dart
import 'Utilisateur.dart';

class Reclamation {
  final int idReclamation;
  final String date;
  final String description;
  final String type;
  final String status;
  final Utilisateur utilisateur;

  Reclamation({
    required this.idReclamation,
    required this.date,
    required this.description,
    required this.type,
    required this.status,
    required this.utilisateur,
  });

  factory Reclamation.fromJson(Map<String, dynamic> json) {
    return Reclamation(
      idReclamation: json['idReclamation'],
      date: json['date'],
      description: json['description'],
      type: json['type'],
      status: json['status'],
      utilisateur: Utilisateur.fromJson(json['utilisateur']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idReclamation': idReclamation,
      'date': date,
      'description': description,
      'type': type,
      'status': status,
      'utilisateur': utilisateur.toJson(),
    };
  }
}
